### ISSUE TITLE
Create new records only after pressing save button

### ISSUE THREAD

**Issue by umulmrum** [chameleon-system/chameleon-system#115]
**Describe the solution you'd like**
When creating a new record in the backend, the record is created as soon as the "New" button is clicked. This leads to empty (and sometimes invalid) records existing in the database as long as the user did not save the record after editing. Also this leads to unclean migration scripts when recorded, as there is an insert operation followed by an update operation.

We should try to change this behavior, so that records are only persisted after the initial save from within the table editor.

**Comment by UlrichKu** [comment]
The entry point for a new entry is \MTTableEditor::Insert(). Being processed in \TCMSTableEditorEndPoint::Insert() and the heavy work done in \TCMSTableEditorEndPoint::_WriteDataToDatabase() and \TCMSTableEditorEndPoint::PostInsertHook() which is especially also called for all the fields in a table.

A copy will start here \MTTableEditor::DatabaseCopy().

Generally the notion of every action in the backend having an id is heavily entrenched in the system. For example the page shown after clicking "New" already wants to show the new entry being loaded from the database by the new id (\MTTableEditor::Init()).

**Comment by bestform** [comment]
As @UlrichKu mentioned, this is unfortunately a concept that cannot be broken easily. I will close this issue for now as a "Won't fix" but be sure that we WILL fix this, if a potential rewrite is on the table.
